server/routes/ad.ts

```diff
@@ -89,7 +89,10 @@ const deleteDetector = async (
     };
   } catch (err) {
     console.log('Anomaly detector - deleteDetector', err);
-    return { ok: false, error: err.body || err.message };
+    return {
+      ok: false,
+      error: getErrorMessage(err),
+    };
```

Comment:
Since `getErrorMessage()` doesn't check for `err.body` at all, seems like this will change the functionality. Can you confirm that the error messages returned here make sense?

Reply:
It is basically same as before. For `err.body || err.message`, if `body` is not there, it will return `message` instead. For `getErrorMessage()` we try to get  `err.body.error.reason`, if `body` is undefined, we change to `message` as well.

The only case is `err.body` is there, but `err.body.error` is undefined. But I have double checked exception log and all of APIs actually have both `err.body.error.reason` and `message`. It should be good for us.

Reply:
Got it, yeah I was thinking that maybe if `err.body` was there, then changing to just use `getErrorMessage()` which only uses `err.message` then we would possibly return a different message. Thanks for checking the apis. Seems that the backend error handling was probably less standardized when first implemented

Reply:
yeah. It might also be caused by our AD ES plugin exception format is not standardized somehow.